packages/graphql-eslint-rules/src/utils.ts

```diff
@@ -0,0 +1,22 @@
+import { join, dirname } from "path";
+import fs, { StatOptions } from "fs";
+
+export function checkDirForPkg(dir: string): string | null {
+  const pathname = join(dir, "package.json");
+  let stat;
+
+  try {
+    stat = fs.statSync(pathname, { throwIfNoEntry: false } as StatOptions);
+  } finally {
+    if (stat && stat.isFile()) {
+      return pathname;
+    }
+  }
```

Comment:
For my learning, what could it still throw for?

Reply:
I tried to run it in TMP and for some reason it threw an error even with `throwIfNotEntry`